docs/contributing/git-style.rst

```diff
@@ -46,3 +46,64 @@ features or plugins to streamline this process (e.g.,
 `magit <https://magit.vc/>`_). Git also provides ``git commit --fixup`` and
 ``git rebase --autosquash`` which can streamline this process.
 
+It is also possible to ask GitHub to squash your commits for you. This is the
+"squash and merge" option when merging your PR.
+
+Commit Message Style
+====================
+
+Commit messages are part of the repository in the same way as the files in a
+repository are. And as developers may use ``grep`` to orient themselves around
+the repository, developers may also use ``git log`` and ``git blame`` to
+understand more of the history of a file and its context. So, it pays to be
+mindful of the content and structure of commits and their messages.
+
+Commit messages should be written with your reviewers in mind as well as future
+developers (or generally the codebase itself). Changes to the repository are
+already included in a commit so use your commit message to focus on things that
+would not readily be apparent from just looking at the textual changes.
+
+Here are some examples of useful things to note for your reviewers and future
+developers:
+
+- Why was this change made?
+
+- What plausible alternative implementations did you consider but rule out?
+
+- What general thing or abstraction is this change in furtherance of?
+
+- If this change fixes a bug, what was the original bug or its symptoms?
+
+- If you did some additional changes along the way to facilitate the main
+  thrust of your work, explain the rational behind any non-obvious but required
+  changes.
+
+At the same time, there is no reason to agonize over the content of a commit
+message or be overly verbose. Applying an automatic formatter changes a lot of
+files but doesn't require a lot of context, while a one-line change might merit
+several paragraphs of context if the rationale is subtle or non-obvious.
+
+The first line of a commit message should be short, generally less than 80
+characters long.
```

Comment:
Could we add the standard language about how this line should be imperative? or do you think that's getting overly prescriptive.

Similar thought: I'm fond of starting this line with some context for what the logical center is (e.g. `csv-import:` `tsuba:` etc.) I don't want to push pedantic requirements on folks, but if other people like that it is nice for messages to have some consistent structure.

Reply:
I second both suggestions of Tyler

Reply:
I wonder about all these Jira tags, e.g., `[KAT-XXX]`, does anyone know if the Jira linking works okay if it is just in the body of the commit rather than in the object line?

Reply:
Checked and it looks like Jira reads commit bodies.